### PR TITLE
Update network policy doc

### PR DESCRIPTION
**Pull request by hibell**
**What this PR does / why we need it?**:

- Add doc for network policy.

**Does this PR introduce a user-facing change?**

- [X] User guide
- [ ] `CHANGELOG.md`

**Which issue(s) this PR fixes**:

Fixes WASdev/websphere-liberty-operator#90.

### CODE REVIEW

**Comment by hibell** [comment]
@leochr The PR has changes for the probe doc updates. I'll fix after the probe doc updates are merged into main.

**Comment by hibell** [comment]
@leochr I rebased to include probe doc updates so PR is good to review now.

File: internal/doc/draft-user-guide-v1.adoc [hunk 1]
```diff
@@ -718,11 +721,96 @@ When `expose` is **not** set to `true`, the Knative service is labeled with `ser
...
+
+OpenShift provides support for network policies by default. Users deploying to other Kubernetes platforms will need to
+ensure that their network plugin supports network policies as described
+link:https://kubernetes.io/docs/concepts/services-networking/network-policies/#prerequisites[here].
+
+==== Default Network Polices
+
+* The default network policy created for applications that are not exposed limits ingress traffic to pods in the same
+namespace that are part of the same application. Traffic is limited to only the ports configured by the service.
+* For exposed applications on OpenShift, the network policy limits ingress traffic from the OpenShift ingress
```
Comment (leochr) [inline internal/doc/draft-user-guide-v1.adoc:750]: "network policy limits ingress traffic from the OpenShift ingress" : traffic could still come from other pods directly, perhaps we should state that the "network policy allows traffic from the OpenShift ingress"

File: internal/doc/draft-user-guide-v1.adoc [hunk 2]
```diff
@@ -718,11 +721,96 @@ When `expose` is **not** set to `true`, the Knative service is labeled with `ser
...
+==== Configuring the Network Policy
+
+The network policy can be configured to allow either certain namespaces and/or pods using
+`.spec.networkPolicy.namespaceLabels` and `.spec.networkPolicy.fromLabels`, respectively.
+
+By default, `.spec.networkPolicy.namespaceLabels` is set to the same namespace that the application is deployed to, and
+`.spec.networkPolicy.fromLabels` is set to select pods that belong to the same application specified by
+`.spec.applicationName`.
+
+Example of allowing ingress traffic from pods in the same namespace labeled with `role: frontend`:
```
Comment (leochr) [inline internal/doc/draft-user-guide-v1.adoc:765]: "from pods labeled with `role: frontend` and are from the same namespace" - so users don't mistakenly think it's referring to namespace labels

File: internal/doc/draft-user-guide-v1.adoc [hunk 3]
```diff
@@ -718,11 +721,96 @@ When `expose` is **not** set to `true`, the Knative service is labeled with `ser
...
+    fromLabels:
+      role: frontend
+----
+
 === Certificates
 
 Specify your own certificates for the Service and Route using fields `.spec.service.certificateSecretRef` and `.spec.route.certificateSecretRef`.
 
-Example of cerificates specified for the Route:
+Example of certificates specified for the Route:
```
Comment (leochr) [inline internal/doc/draft-user-guide-v1.adoc:813]: good catch

**Review by leochr — COMMENTED** [review]
@hibell Looks great! Added some minor comments.

**Review by leochr — APPROVED** [review]
@hibell Thanks for the update. Looks good.

**Pull request merged**